cardano-api/src/Cardano/API.hs

```diff
@@ -0,0 +1,286 @@
+module Cardano.API (
+    -- * Eras
+    Byron,
+    Shelley,
+    HasTypeProxy(..),
+    AsType(..),
+    -- * Cryptographic key interface
+    -- $keys
+    Key,
+    VerificationKey(..),
+    SigningKey(..),
+    getVerificationKey,
+    verificationKeyHash,
+    castVerificationKey,
+    castSigningKey,
+
+    -- ** Generating keys
+    generateSigningKey,
+    deterministicSigningKey,
+    deterministicSigningKeySeedSize,
+
+    -- ** Hashes
+    -- | In Cardano most keys are identified by their hash, and hashes are
+    -- used in many other places.
+    Hash(..),
+    castHash,
+
+    -- * Payment addresses
+    -- | Constructing and inspecting normal payment addresses
+    Address(..),
+    NetworkId(..),
+    -- * Byron addresses
+    makeByronAddress,
+    ByronKey,
+    -- * Shelley addresses
+    makeShelleyAddress,
+    PaymentCredential(..),
+    StakeAddressReference(..),
+    PaymentKey,
+    PaymentExtendedKey,
+
+    -- * Stake addresses
+    -- | Constructing and inspecting stake addresses
+    StakeAddress(..),
+    StakeCredential(..),
+    makeStakeAddress,
+    StakeKey,
+    StakeExtendedKey,
+
+    -- * Building transactions
+    -- | Constructing and inspecting transactions
+    TxBody(..),
```

Comment:
We need to review all these (..) exports and not include the ones that are Byron/Shelley specific constructors. Only the era-independent constructors should be exported here.

Reply:
@dcoutts Do I need to also export all functions regardless of what era they refer to?

Reply:
All the "public looking" functions are fine, even when they're era specific. It's the constructors that reveal the types from the underlying libs that we want to not expose, since the point of that top level module is to provide a full wrapper and not expose the underlying types.